clang/test/CheckedC/inferred-bounds/widened-bounds.c

```diff
@@ -52,3 +52,21 @@ void f3() {
 // CHECK: [B1]
 // CHECK-NOT: upper_bound(p) = 1
 }
+
```

Comment:
Could you add tests of widening by more than 1 element?   
```
if (p[0]) {
  if (p[1]) {
    ... // bounds should be widened to p + 2 here.
```
I know that currently the compiler will issue an error for p[2] because it'll think the access is out-of-bounds.   You can use the `-verify` command-line option to clang to handle this.  It'll let you check that the dataflow analysis produces the right result, even if the compiler still produces an error later.